metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertCommentCodeAction.scala

```diff
@@ -0,0 +1,170 @@
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+
+    (for {
+      content <- buffers.get(path)
+      tokens <- tokenizeIfNotRangeSelection(range, content)
+      codeAction <- createActionIfPossible(
+        path,
+        tokens,
+        range,
+        isSingleLineComment(content),
+      )
+    } yield codeAction).toList
+  }
+
+  private def tokenizeIfNotRangeSelection(range: l.Range, content: String) = {
+    Option
+      .when(range.getStart == range.getEnd)(
+        Trees.defaultTokenizerDialect(content).tokenize
+      )
+      .flatMap(_.toOption)
+  }
+
+  private def isSingleLineComment(content: String)(
+      t: Token
+  ): Boolean = t match {
+    case tc: Token.Comment =>
+      val currentLine = content.split('\n')(t.pos.startLine)
+      tc.pos.startLine == tc.pos.endLine &&
+      currentLine.slice(tc.pos.startColumn, tc.pos.startColumn + 2) == "//"
+    case _ => false
+  }
+
+  private def createActionIfPossible(
+      path: AbsolutePath,
+      tokens: Tokens,
+      range: l.Range,
+      isSingleLineComment: Token => Boolean,
+  ): Option[l.CodeAction] = {
+    val indexOfLineTokenUnderCursor = tokens.lastIndexWhere(t =>
+      t.pos.startLine == range.getStart.getLine
+        && t.pos.endLine == t.pos.startLine
+        && t.pos.startColumn < range.getStart.getCharacter
+        && isSingleLineComment(t)
+    )
+    if (indexOfLineTokenUnderCursor != -1) {
+      // tokens that are strictly before cursor, i.e. they end before cursor position
+      val tokensBeforeCursor = tokens.take(indexOfLineTokenUnderCursor)
+      // token under the cursor + following tokens
+      val tokensAfterCursor = tokens.drop(indexOfLineTokenUnderCursor)
+      val textEdit = createTextEdit(
+        tokensBeforeCursor,
+        tokensAfterCursor,
+        range,
+      )
+      Some(
+        CodeActionBuilder.build(
+          title = ConvertCommentCodeAction.Title,
+          kind = this.kind,
+          changes = List(path -> List(textEdit)),
+        )
+      )
+    } else {
+      None
+    }
+  }
+
+  private def createTextEdit(
+      tokensBeforeCursor: Tokens,
+      tokensAfterCursor: Tokens,
+      range: l.Range,
+  ) = {
+    val commentBeforeCursor = collectContinuousComments(
+      tokens = tokensBeforeCursor.reverse
+    ).reverse
+
+    val commentAfterCursor = collectContinuousComments(
+      tokens = tokensAfterCursor
+    )
+    val commentStart = commentBeforeCursor.headOption
+      .map(_.pos.toLsp.getStart)
+      .getOrElse(range.getStart)
+    val commentEnd = commentAfterCursor.lastOption
+      .map(_.pos.toLsp.getEnd())
+      .getOrElse(range.getEnd())
+
+    val commentTokens = commentBeforeCursor ++ commentAfterCursor
+    val replaceText =
+      commentTokens
+        .map(_.value.trim())
```

Comment:
```suggestion
        .map(_.value.strip())
```
Nit, maybe we could use a newer method that handles Unicode whitespace. :slightly_smiling_face:

Reply:
Nice catch, I didn't know about `strip`!

Reply:
That makes jdk 8 sad.

Reply:
Oh, you're right, I completely forgot that Metals still supports Java 8, sorry for the trouble! :facepalm: Maybe now that Java 21 LTS is released it would be a good time to think about moving on to 11+. :wink:

Reply:
No worries, reverted to `trim` and added comment.